addons/l10n_mx/models/account.py

```diff
@@ -48,7 +48,8 @@ def mx_search_tags(self, code):
     def _onchange_code(self):
         if self.company_id.country_id.id == self.env.ref('base.mx').id and self.code:
             tags = self.mx_search_tags(self.code)
-            self.tag_ids = tags
+            self.tag_ids = self.tag_ids.filtered(
```

Comment:
Use `set` operators instead of `set` methods.
I mean: `self.tag_ids = self.tag_ids.filtered(lambda r: r.color != 4) | tags` (Notice the pipe `|` instead of `concat`)